table/block_based_table_reader.cc

```diff
@@ -1119,10 +1119,16 @@ bool BlockBasedTable::PrefixMayMatch(const Slice& internal_key) {
     return true;
   }
 
+  ParsedInternalKey parsed_internal_key;
+  if (!ParseInternalKey(internal_key, &parsed_internal_key)) {
+    // This shouldn't happen, but if it does just stumble forward.
+    return true;
+  }
+
   assert(rep_->ioptions.prefix_extractor != nullptr);
   auto prefix = rep_->ioptions.prefix_extractor->Transform(
-      ExtractUserKey(internal_key));
-  InternalKey internal_key_prefix(prefix, 0, kTypeValue);
+      parsed_internal_key.user_key);
+  InternalKey internal_key_prefix(prefix, parsed_internal_key.sequence, kTypeValue);
```

Comment:
Both way works for me.

By the way, I encourage you to use full bloom filters, instead of block based bloom filters. (See the second parameter in https://github.com/facebook/rocksdb/blob/v4.1/include/rocksdb/filter_policy.h#L128-L129 ) It performs better.